client/my-sites/plugins/controller.js

```diff
@@ -147,3 +147,28 @@ export function scrollTopIfNoHash( context, next ) {
 	}
 	next();
 }
+
+export function redirectOnUnknownCategory( context, next ) {
+	const category = context.params.category;
+	const site = context.params.site;
+
+	if ( ! ALLOWED_CATEGORIES.includes( category ) ) {
+		const redirectionUrl = `/plugins/${ site || '' }`;
+		page.redirect( redirectionUrl );
+	}
+
+	next();
+}
```

Comment:
This might be in conflict with the generic category links @Automattic/bespin is adding to product details.
![Screenshot 2022-08-24 at 15-52-13 Booking Calendar Plugin ‹ Business - Simple Site — WordPress com](https://user-images.githubusercontent.com/811776/186545216-32e2908b-267c-4f8b-bcce-026e99808f3b.png)

![Screenshot 2022-08-24 at 15-52-04 Plugins ‹ Business - Simple Site — WordPress com](https://user-images.githubusercontent.com/811776/186545182-00514ada-e5e6-45aa-a8ba-b1ba7f1f4b6a.png)

Reply:
I didn't know about the generic categories and this change will affect this, I also remove it

Reply:
I just found them yesterday by accident while testing the search box changes https://github.com/Automattic/wp-calypso/issues/66899